From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Wed, 30 Dec 2015 18:54:56 +0000
Subject: Fix Save-As functionality.
X-Git-Url: http://quickgit.kde.org/?p=libkeduvocdocument.git&a=commitdiff&h=8c8cf0290332c0c6c0bf5b1c7248ff066aa25022
---
Fix Save-As functionality.

File system operations like QFile require paths instead of URLs, which
is the difference between the uses of:
* url.toDisplayString -> file:///the/path
* url.toLocalFile -> /the/path
Correctly use toLocalFile for all file system calls.

BUG: 356233
(cherry picked from commit 0cb9b9aea2264fc66d25fb6d93adc32c75dfef97)
---


--- a/keduvocdocument/keduvocdocument.cpp
+++ b/keduvocdocument/keduvocdocument.cpp
@@ -339,7 +339,7 @@
         || ( d->m_autosave->managedFile() != tmp ) ) {
 
         autosave = new KAutoSaveFile;
-        ErrorCode autosaveError = d->initializeKAutoSave( *autosave,  tmp.toDisplayString(), flags );
+        ErrorCode autosaveError = d->initializeKAutoSave( *autosave, tmp.toLocalFile(), flags );
         if ( autosaveError != NoError) {
             delete autosave;
             return autosaveError;
@@ -355,7 +355,7 @@
         }
     }
 
-    QFile f( tmp.toDisplayString() );
+    QFile f( tmp.toLocalFile() );
     if ( !f.open( QIODevice::WriteOnly ) ) {
         qCritical() << i18n( "Cannot write to file %1", f.fileName() );
         return FileCannotWrite;
